app/uk/gov/hmrc/tariffclassificationfrontend/service/CasesService.scala

```diff
@@ -230,6 +231,15 @@ class CasesService @Inject()(appConfig: AppConfig,
     connector.findCasesByQueue(queue, pagination)
   }
 
+  def countCasesByQueue(operator: Operator)(implicit hc: HeaderCarrier): Future[Map[String, Int]] = {
+    for {
+      countMyCases <- getCasesByAssignee(operator, NoPagination())
+      countByQueue <- reportingService.getQueueReport
+    } yield countByQueue.map(r => (
+          r.group.getOrElse(Queues.gateway.id), r.value.size)).toMap ++
+          Map("my-cases" -> countMyCases.size)
+  }
```

Comment:
Would it be a bit more readable if the mapping to final result was completed inside the for comprehension? Something like this:
def countCasesByQueue(operator: Operator)(implicit hc: HeaderCarrier): Future[Map[String, Int]] = {
    for {
      countMyCases <- getCasesByAssignee(operator, NoPagination())
      countByQueue <- reportingService.getQueueReport
      queueIdAndCount = countByQueue.map(r => (
        r.group.getOrElse(Queues.gateway.id), r.value.size)).toMap + ("my-cases" -> countMyCases.size)
    } yield queueIdAndCount
  }

Reply:
This one already looks good to me, actually the name of the actual method (**countCasesByQueue**) should be like the new var you have created (**queueIdAndCount**). Maybe is that scala madness...